runtime/lua/vim/diagnostic.lua

```diff
@@ -1577,6 +1581,12 @@ M.handlers.virtual_text = {
 
     local virt_text_ns = ns.user_data.virt_text_ns
     local buffer_line_diagnostics = diagnostic_lines(diagnostics)
+
+    if opts.virtual_text.current_line == true then
+      local lnum = api.nvim_win_get_cursor(0)[1] - 1
+      buffer_line_diagnostics = { [lnum] = buffer_line_diagnostics[lnum] }
+    end
```

Comment:
Is it worth moving the call to `diagnostic_lines` into the `else` branch of this condition so that we don't throw away a bunch of data we don't need? In the `then` branch we'd need to compute the buffer line diagnostics "manually", but that is easy (simple for loop + table.insert).

Reply:
Good idea!